pandas/_libs/tslibs/offsets.pyx

```diff
@@ -361,6 +370,58 @@ class _BaseOffset(object):
         out = '<%s' % n_str + className + plural + self._repr_attrs() + '>'
         return out
 
+    def __setstate__(self, state):
+        """Reconstruct an instance from a pickled state"""
```

Comment:
no just define dunder reduce
you can’t use getstate/setstate in cython class

Reply:
I'll try again with `__reduce__`.  Don't hold your breath.

If we define `__setstate__` in the non-cython class, trying to set self.n raises an AttributeError because it is readonly.

Reply:
Isn't `__reduce__` for pickling?  The problem here is un-pickling legacy offsets.